UPGRADING.md

```diff
@@ -0,0 +1,164 @@
+- Kwil CLI, GO SDK, and JS SDK have been updated to use the JSON-RPC server.
+
+#### Breaking Changes:
+
+- The HTTP server (`http_listen_addr`) is DEPRECATED and will be removed in the susbequent release.
+- The gRPC server (`grpc_listen_addr`) is no longer exposed.
+
+### Node Configuration Changes
+
+In the node's `config.toml` file, the following changes have been made:
+
+- Added `app.rpc_timeout`, which imposes a timeout on RPC requests. The default is 45 seconds.
+- Added `app.db_read_timeout`, which imposes a timout on read-only DB transactions (i.e. actions/procedures with a [`view` tag](https://docs.kwil.com/docs/kuneiform/supported-features#access-tags) or an ad-hoc SELECT query). The default 5 seconds.
+- Added `rpc.broadcast_tx_timout`, which imposes a timeout for awaiting transaction confirmation when using the `--sync` flag in `kwil-cli`. The default is 10 seconds.
+- Added `app.snapshots` to support snapshot creation. See the `StateSyncConfig` struct for more information.
+- Added `app.snapshot_file` to suport starting a network with a state at genesis. This is to support the Network Migrations feature.
+
+### CLI Changes
+
+- The `kwil-cli` has been updated to use the JSON-RPC server.
+- The `--kwil-provider` flag is DEPRECATED and will be removed in the subsequent release. Use the `--provider` flag instead.
+- Added `kwil-cli utils parse` to parse a Kuneiform file and output the schema as JSON.
+- Added `kwil-cli utils decode-tx` to decode a transaction from a base64 string.
+
+#### Breaking Changes
+
+- Because Kwil-CLI now communicates over JSON-RPC, you should pass the JSON-RPC listen address to the CLI with the `--provider` flag or `kwil-cli configure` command.
+- For uncomfirmed transactions, `kwild` now returns `null` for the `tx` field. Using older versions of `kwil-cli` to query unconfirmed transactions may have unexpected behavior.
+
+### Kwil Admin
+
+- Added `kwil-admin snapshot create` command for creating network snapshots (both CometBFT and Postgres data).
+
+#### Breaking Changes
+
+- The admin service now listens on `"/tmp/kwild.socket"` by default. Previously, the admin service listened on `"unix:///tmp/kwil_admin.sock"`.
+
+### Core Module (Go SDK)
+
+This section covers changes to the `core` module, which provides the SDK and client for Kwil. If you do not use the Go SDK, you can skip this section. Other Kwil tooling (e.g. CLI, JS SDK, etc.) account for these changes.
+
+The `core` module is tagged as `core/v0.2.0` for the Kwil DB v0.8 release.
+
+#### `core/crypto/auth`
+
+##### Breaking Changes
+
+- The JSON tags on the `Signature` struct have changed:
+
+```go
+type Signature struct {
+    Signature []byte `json:"sig"` // was `json:"signature_bytes"`
+    Type string `json:"type"` // was `json:"signature_type"`
+}
+```
+
+#### `core/types`
+
+- The `Schema` struct and all of it's composing types are now mirrored in `core/types/transactions`.
```

Comment:
related, but we used to keep the main schema struct in `common.Schema`. It is now in `core/types.Schema`